tribe-ext-select-ticket-providers/index.php

```diff
@@ -0,0 +1,132 @@
+ * Extension Class: Tribe__Extension__Select_Ticket_Providers
+ * Author:          Modern Tribe, Inc.
+ * Author URI:      http://m.tri.be/1971
+ * License:         GPLv2 or later
+ * License URI:     https://www.gnu.org/licenses/gpl-2.0.html
+ */
+
+// Do not load directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	die( '-1' );
+}
+
+// Do not load unless Tribe Common is fully loaded.
+if ( ! class_exists( 'Tribe__Extension' ) ) {
+	return;
+}
+
+/**
+ * Extension main class, class begins loading on init() function.
+ */
+class Tribe__Extension__Select_Ticket_Providers extends Tribe__Extension {
+
+	/**
+	 * Contains the full list of providers from before we removed any
+	 *
+	 * @var array
+	 */
+	protected $default_providers;
+
+	/**
+	 * Option key for which providers are enabled
+	 *
+	 * @var string
+	 */
+	protected $option_key_enabled_providers = 'tickets_enabled_providers';
+
+	/**
+	 * Setup the Extension's properties.
+	 *
+	 * This always executes even if the required plugins are not present.
+	 */
+	public function construct() {
+		$this->add_required_plugin( 'Tribe__Tickets__Main' );
+		$this->add_required_plugin( 'Tribe__Tickets_Plus__Main' );
+
+		$this->set_url( 'https://theeventscalendar.com/extensions/select-ticket-providers' );
+	}
+
+	/**
+	 * Extension initialization and hooks.
+	 */
+	public function init() {
+		add_action( 'init', array( $this, 'remove_providers' ), 100 );
+		add_action( 'admin_init', array( $this, 'add_settings' ) );
+	}
+
+	/**
+	 * Add settings to tribe settings page
+	 */
+	public function add_settings() {
+		require_once dirname( __FILE__ ) . '/src/Tribe/Settings_Helper.php';
+
+		// Setup fields on the settings page
+		$setting_helper = new Tribe__Settings_Helper();
+		$providers = $this->get_default_providers();
+
+		$setting_helper->add_field(
+			$this->option_key_enabled_providers,
+			array(
+				'type'            => 'checkbox_list',
+				'label'           => esc_html__( 'Ticket Providers', 'tribe-extension' ),
+				'tooltip'         => esc_html__( 'Select which ticket providers you want available on your site.', 'tribe-extension' ),
+				'default'         => array_keys( $providers ),
+				'validation_type' => 'options_multi',
+				'options'         => $providers,
+			),
+			'event-tickets',
+			'ticket-enabled-post-types',
+			false
+		);
+	}
+
+	/**
+	 * Gets defaults providers
+	 *
+	 * This should be called prior to altering the list any.
+	 */
+	protected function get_default_providers() {
+		if ( ! isset( $this->default_providers ) ) {
+			$this->default_providers = Tribe__Tickets__Tickets::modules();
+		}
+
+		return $this->default_providers;
+	}
+
+	/**
+	 * Removes ticket providers specified in the options
+	 */
+	public function remove_providers() {
+		// The meat of this extension is a hack, that requires certain versions of PHP.
+		// @todo Once we get a proper filter for the ticket providers we can remove the hack.
+		if ( version_compare( PHP_VERSION, '5.3.2', '<' ) ) {
```

Comment:
You're right! `ReflectionProperty::setAccesible()` only requires 5.3.0. `ReflectionMethod::setAccesible()` if the one that requires 5.3.2. I'll make the adjustment.